cibuildwheel/__main__.py

```diff
@@ -402,6 +402,15 @@ def get_build_identifiers(
 def detect_warnings(*, options: Options, identifiers: Iterable[str]) -> list[str]:
     warnings = []
 
+    python_version_deprecation = ((3, 11), 3)
+    if sys.version_info[:2] < python_version_deprecation[0]:
+        python_version = ".".join(map(str, python_version_deprecation[0]))
+        msg = (
+            f"cibuildwheel {python_version_deprecation[1]}+ will require Python {python_version}+, "
+            "please upgrade your python version."
+        )
+        warnings.append(msg)
```

Comment:
warnings are just printed (i.e. not using `log.warning`).
This might end up with warnings not being seen easily (mostly on GHA where warnings are reported outside of the logs themselves).
Should `print_preamble` be updated to use `log.warning` ? (or add the correct prefix for GHA) ?

Reply:
Done.
It used to look like:
```
...
 test_requires: 
    -r
    requirements-test.txt

Cache folder: /Users/Matt/Library/Caches/cibuildwheel

Warnings:
  cibuildwheel 3+ will require Python 3.11+, please upgrade your python version.
  test_command: '{python}' and '{pip}' are no longer needed, and will be removed in a future release. Simply use 'python' or 'pip' instead.
  before_build: '{python}' and '{pip}' are no longer needed, and will be removed in a future release. Simply use 'python' or 'pip' instead.

Here we go!
```

It now looks like:
```
...
  test_requires: 
    -r
    requirements-test.txt

Cache folder: /Users/Matt/Library/Caches/cibuildwheel

Warning: cibuildwheel 3+ will require Python 3.11+, please upgrade your python version.

Warning: test_command: '{python}' and '{pip}' are no longer needed, and will be removed in a future release. Simply use 'python' or 'pip' instead.

Warning: before_build: '{python}' and '{pip}' are no longer needed, and will be removed in a future release. Simply use 'python' or 'pip' instead.

Here we go!
```

Reply:
If you want to change it while touching this, we could say it will be removed in version 3. In version 3, we can detect this and throw an error instead, perhaps?

Reply:
> In version 3, we can detect this and throw an error instead, perhaps?

`requires-python` should prevent such a thing ?
Maybe it could  happen in some edge cases but given it's checked after all option parsing, I guess it would fail before the version check.

Reply:
Oh, sorry, I meant the `'{python}' and '{pip}' are no longer needed` part.

Reply:
It makes much more sense now.
Yes we can update those to say it'll be remove in cibuildwheel 3 while we're at it. It's probably a good thing to error out in 3.x for a while at least before removing the check (if we remove it).